### PR TITLE
Address tests

### PR DESCRIPTION
**Pull request by shanemc-goog**
### Description
Add focused tests for Address type. Two tests are commented out and for issues that should be fixed.

### Checklist
- [x] Created tests which fail without the change (if possible)

### CODE REVIEW

File: browser-test/src/currency.test.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import {AdminPrograms, AdminQuestions, ApplicantQuestions, loginAsAdmin, loginAsTestUser, logout, selectApplicantLanguage, startSession, resetSession} from './support'
```
Comment (shanemc-goog) [inline browser-test/src/currency.test.ts:1]: I don't know why this file is showing up, it's from a committed PR.

File: browser-test/src/address.test.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        await selectApplicantLanguage(pageObject, 'English');
+
+        await applicantQuestions.applyProgram(programName);
+        await applicantQuestions.answerAddressQuestion('', '', '', '', '', 1);
+        await applicantQuestions.clickNext();
+
+      });
+
+      it('does not submit', async () => {
+        // Second question hos errors.
```
Comment (dkatzz) [inline browser-test/src/address.test.ts:214]: has* also

File: browser-test/src/address.test.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+      await selectApplicantLanguage(pageObject, 'English');
+
+      await applicantQuestions.applyProgram(programName);
+      await applicantQuestions.answerAddressQuestion('1234 St', 'Unit B', 'Sim', 'Ames', '54321');
+      await applicantQuestions.clickNext();
+
+      await applicantQuestions.submitFromReviewPage(programName);
+    });
+
+    it('with invalid address does not submit', async () => {
```
Comment (dkatzz) [inline browser-test/src/address.test.ts:58]: Maybe change to "empty address" and add another test for an invalid address - i.e. incorrect zip code.

File: browser-test/src/address.test.ts [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+      let error = await pageObject.$('.cf-address-street-1-error >> nth=0');
+      expect(await error.isHidden()).toEqual(false);
+      error = await pageObject.$('.cf-address-city-error >> nth=0');
+      expect(await error.isHidden()).toEqual(false);
+      error = await pageObject.$('.cf-address-state-error >> nth=0');
+      expect(await error.isHidden()).toEqual(false);
+      error = await pageObject.$('.cf-address-zip-error >> nth=0');
+      expect(await error.isHidden()).toEqual(false);
+
+      // Second question hos no errors.
```
Comment (dkatzz) [inline browser-test/src/address.test.ts:127]: has*

File: browser-test/src/address.test.ts [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+      let error = await pageObject.$('.cf-address-street-1-error >> nth=0');
+      expect(await error.isHidden()).toEqual(true);
+      error = await pageObject.$('.cf-address-city-error >> nth=0');
+      expect(await error.isHidden()).toEqual(true);
+      error = await pageObject.$('.cf-address-state-error >> nth=0');
+      expect(await error.isHidden()).toEqual(true);
+      error = await pageObject.$('.cf-address-zip-error >> nth=0');
+      expect(await error.isHidden()).toEqual(true);
+
+      // Second question hos errors.
```
Comment (dkatzz) [inline browser-test/src/address.test.ts:158]: Same here :)

File: browser-test/src/address.test.ts [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+        expect(await error.isHidden()).toEqual(false);
+        error = await pageObject.$('.cf-address-state-error >> nth=1');
+        expect(await error.isHidden()).toEqual(false);
+        error = await pageObject.$('.cf-address-zip-error >> nth=1');
+        expect(await error.isHidden()).toEqual(false);
+      });
+
+      // TODO: Fix code so this test passes.
+      /*
+      it('optional has no errors', async () => {
```
Comment (dkatzz) [inline browser-test/src/address.test.ts:227]: I thought optional only shows errors if the required one does. This one fails with optional being the only empty question?

File: browser-test/src/currency.test.ts (hunk 1 above)
Comment (dkatzz) [reply browser-test/src/currency.test.ts:1]: Hmmm yeah that's weird! Did you merge in main?

**Review by dkatzz — COMMENTED** [review]
Thanks for adding these :)

File: browser-test/src/address.test.ts (hunk 6 above)
Comment (shanemc-goog) [reply browser-test/src/address.test.ts:227]: This is the case where if the required one has an error, the empty-optional shows error messages.  

Are you saying that's WAI even though optionals don't have to be specified?

Comment (dkatzz) [reply browser-test/src/address.test.ts:227]: Oops, I misread it - empty-optional shouldn't show errors in this case. We can uncomment it once https://github.com/seattle-uat/civiform/issues/1652 is resolved.

File: browser-test/src/currency.test.ts (hunk 1 above)
Comment (shanemc-goog) [reply browser-test/src/currency.test.ts:1]: I'm pretty sure I did... It seems to all work out based on previous instance of this though...

File: browser-test/src/address.test.ts (hunk 2 above)
Comment (shanemc-goog) [reply browser-test/src/address.test.ts]: done

File: browser-test/src/address.test.ts (hunk 5 above)
Comment (shanemc-goog) [reply browser-test/src/address.test.ts]: done

File: browser-test/src/address.test.ts (hunk 6 above)
Comment (shanemc-goog) [reply browser-test/src/address.test.ts:239]: added issue references in the comment.

**Review by dkatzz — APPROVED** [review]

**Pull request merged**